core/uemis-downloader.c

```diff
@@ -198,6 +198,7 @@ static void uemis_get_weight(char *buffer, weightsystem_t *weight, int diveid)
 		lbs_to_grams(ascii_strtod(buffer, NULL)) :
 		lrint(ascii_strtod(buffer, NULL) * 1000);
 	weight->description = translate("gettextFromC", "unknown");
+	weight->auto_filled = false;
```

Comment:
What about either
```
    *weight = empty_weightsystem;
```
at the top of this function
or
```
    weightsystem_t ws = empty_weightsystem;
```
where it is generated?
That would avoid future versions of this bug in case more fields are added.

PS: `uemis_get_weight_unit` could be made local.